src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/WindowBackdrop.cs

```diff
@@ -0,0 +1,404 @@
+
+            case WindowBackdropType.Mica:
+                return ApplyDwmwWindowAttrubute(hWnd, Dwmapi.DWMSBT.DWMSBT_MAINWINDOW);
+
+            case WindowBackdropType.Acrylic:
+                return ApplyDwmwWindowAttrubute(hWnd, Dwmapi.DWMSBT.DWMSBT_TRANSIENTWINDOW);
+
+            case WindowBackdropType.Tabbed:
+                return ApplyDwmwWindowAttrubute(hWnd, Dwmapi.DWMSBT.DWMSBT_TABBEDWINDOW);
+        }
+
+        return ApplyDwmwWindowAttrubute(hWnd, Dwmapi.DWMSBT.DWMSBT_DISABLE);
+    }
+
+    /// <summary>
+    /// Tries to remove backdrop effects if they have been applied to the <see cref="Window"/>.
+    /// </summary>
+    /// <param name="window">The window from which the effect should be removed.</param>
+    internal static bool RemoveBackdrop(System.Windows.Window window)
+    {
+        if (window is null)
+        {
+            return false;
+        }
+
+        IntPtr windowHandle = new WindowInteropHelper(window).Handle;
+
+        return RemoveBackdrop(windowHandle);
+    }
+
+    /// <summary>
+    /// Tries to remove all effects if they have been applied to the <c>hWnd</c>.
+    /// </summary>
+    /// <param name="hWnd">Pointer to the window handle.</param>
+    internal static bool RemoveBackdrop(IntPtr hWnd)
+    {
+        if (hWnd == IntPtr.Zero)
+        {
+            return false;
+        }
+
+        _ = RestoreContentBackground(hWnd);
+
+        if (hWnd == IntPtr.Zero)
+        {
+            return false;
+        }
+
+        if (!NativeMethods.IsWindow(hWnd))
+        {
+            return false;
+        }
+
+        var pvAttribute = 0; // Disable
+        var backdropPvAttribute = (int)Dwmapi.DWMSBT.DWMSBT_DISABLE;
+
+        _ = Dwmapi.DwmSetWindowAttribute(
+            hWnd,
+            Dwmapi.DWMWINDOWATTRIBUTE.DWMWA_MICA_EFFECT,
+            ref pvAttribute,
+            Marshal.SizeOf(typeof(int))
+        );
+
+        _ = Dwmapi.DwmSetWindowAttribute(
+            hWnd,
+            Dwmapi.DWMWINDOWATTRIBUTE.DWMWA_SYSTEMBACKDROP_TYPE,
+            ref backdropPvAttribute,
+            Marshal.SizeOf(typeof(int))
+        );
+
+        return true;
+    }
+
+    /// <summary>
+    /// Tries to remove background from <see cref="Window"/> and it's composition area.
+    /// </summary>
+    /// <param name="window">Window to manipulate.</param>
+    /// <returns><see langword="true"/> if operation was successful.</returns>
+    internal static bool RemoveBackground(System.Windows.Window window)
+    {
+        if (window is null)
+        {
+            return false;
+        }
+
+        // Remove background from visual root
+        window.SetCurrentValue(System.Windows.Controls.Control.BackgroundProperty, Brushes.Transparent);
+
+        IntPtr windowHandle = new WindowInteropHelper(window).Handle;
+
+        if (windowHandle == IntPtr.Zero)
+        {
+            return false;
+        }
+
+        var windowSource = HwndSource.FromHwnd(windowHandle);
```

Comment:
We can get the `windowSource` from `HwndSource.FromVisual` directly without WindowInteropHelper

Reply:
There is `PresentationSource.FromVisual` which you would have to cast to `HwndSource`. It also does more work, `WindowInteropHelper` seems much simpler and this is interop, after all.